lib/if.c

```diff
@@ -1182,6 +1189,18 @@ void if_link_params_free(struct interface *ifp)
 
 /* ----------- CLI commands ----------- */
 
+void interface_xpath(char *xpath, const char *ifname, const char *vrfname)
+{
+	if (vrf_is_backend_netns())
+		snprintf(xpath, XPATH_MAXLEN,
+			 "/frr-interface:lib/interface[name='%s:%s']",
```

Comment:
What does this hack buy us exactly? My understanding is that the interface xpath still needs to be updated whenever it's moved to a different VRF.

Reply:
The problem I am trying to solve is the following:
- you enter the interface configuration node
- someone changes the VRF of the interface
- you continue configuring the interface

You won't be able to continue, because the XPath stored in the `vty` doesn't correspond with the actual state of the system anymore. You have to exit and re-enter the interface node. For example:
```
nfware# conf
nfware(config)# interface enp2s0

<-- move the enp2s0 to a different VRF -->

nfware(config-if)# ip router isis 1
% Failed to get iface dnode in candidate DB
```

By removing the VRF from the interface key we can eliminate this issue. The only problem is that we can't do this for the netns backend because it's possible to have interfaces with the same name in different VRFs. That's why I'm prepending the VRF name to the interface name in the netns case.

Reply:
lets discuss about this..

Reply:
One other way to fix this would be to iterate over all active CLI sessions and update `vty->xpath` whenever an interface is moved to a different VRF (or when a VRF is renamed). A bit clumsy, but would work for all VRF backends.

Reply:
I don't agree with this approach. I even think that the current approach is incorrect. We shouldn't change the config when we move the interface between VRFs. The VRF is part of the key, and we shouldn't change it when the operational state changes. 

I don't like having VRF in the interface key, and it's not needed for the VRF-lite backend, so I prefer to remove it and make things easier and less error-prone. I don't really care much about netns, so I'm just going to keep the current hack for the netns backend.

Reply:
For netns case, that would mean that the xpath interface name contains "<vrf>:<interface>". I try to imagine if I want to inject via grpc some ospf config. That means, I have to provision that xpath combo.
From user perspective, the way to configure will change whether netns or vrflite.

I was thinking of creating a new interface with new settings. Actually, 2 objects would be created:
- the first iface (name=<name>, vrf=<oldvrf>)
- the second iface (name=<name>, vrf=<newvrf>)

Reply:
Yes, you're right. I agree that two objects should be created in this case. The problem is that it doesn't work like this right now. Instead of having two different objects for different VRFs, we're moving one object between VRFs.

I pushed the full implementation of my current approach. But now I am thinking of maybe implementing two different YANG models for netns and VRF-lite backends. The first will include VRF as a key, the second will not.

Reply:
I am not very confortable with the second proposal neither.
if you maintain a specific yang model for netns, xpath ifname will be <vrfname>:<ifname>.

is not there any constraint regarding the length of the xpath ifname ?
If I follow a very big ifname in yang model with vrflite, if I want to apply this to netns mode, will not I have to change the length of the initial ifname length?

Reply:
It's not only an inconvenience. It makes it impossible to implement the configuration of the interface's VRF from FRR.

If the VRF is a leaf, then we just need to implement a `modify` callback that changes the VRF through netlink (in zebra, for example). I don't see any way of implementing it if the VRF is part of the key.

Regarding the netns and other L2-backends, I think we shouldn't treat them as VRF-backends at all. It should be a separate entity. netns (or any other L2-backend) is the first level of virtualization, VRF-lite is the second level - we can use them together. To better understand the idea, please, take a look at RFCs for [Network Instances](https://www.rfc-editor.org/rfc/rfc8529) and [Logical Network Elements](https://www.rfc-editor.org/rfc/rfc8530.html). IMO we should treat netns as an LNE and VRF-lite as a NI.

P.S. I am closing this PR as we're going with your idea of updating the vty xpaths.